### ISSUE TITLE
[Feature Request] Keymap-specific commenting

### ISSUE THREAD

**Issue by sharpchen** [numToStr/Comment.nvim#484]
## Motivation

I've been using [vitepress](https://vitepress.dev/), it has a feature called [line highlighting](https://vitepress.dev/guide/markdown#line-highlighting-in-code-blocks) which allows to highlight lines in markdown code block with certain tailing comment like:

```js
export default {
  data () {
    return {
      msg: 'Removed' // [!code --]
      msg: 'Added' // [!code ++]
    }
  }
}
```

So I'd like to have some keymaps for easily appending tailing comments.
Since we are working on markdown file, so it should do commenting base on the code context in code block.
In other words, for c-like languages it should add `// [!code ++]`, for others like lua should be `-- [!code ++]`

```lua
vim.keymap.set('x', 'gva', function() 
  --[[ add `[!code ++]` after each line base on context ]] 
end)
```

I've searched the codebase but couldn't find out a solution with current api.
If there's possible workaround, please tell me how to.

## Extra question

It seems context-based commenting is available, I can toggle comment in markdown code block.
But for some languages like `cs` and `fs`, it's not commenting as expected, is it a bug?

**Comment by sharpchen** [comment]
Found a workaround, ~~unfortunately there's no api to toggle insert and comment insert for selection.~~

```lua
---@return string?
local function get_lang()
  local curline = vim.fn.line('.')
  local lang = vim.treesitter.get_parser():language_for_range({ curline, 0, curline, 0 }):lang()
  return lang
end

---@return string | string[]
---@return string
local function get_ctx()
  local lang = get_lang()
  assert(lang, 'lang not found')
  local cs = require('Comment.ft').get(lang)
  assert(cs, 'comment string not found')
  return cs, lang
end

---@param suffix string
---@return function
local function comment_eol(suffix)
  return function()
    local success, cs, lang = pcall(get_ctx)
    if not success then
      vim.notify('getting context failed', vim.log.levels.ERROR)
      return
    end
    local ori_cs = type(cs) == 'string' and cs or cs[1]

    local pos = ori_cs:find('%%s')
    local sub = pos and ori_cs:sub(1, pos - 1) .. ' ' .. suffix .. ori_cs:sub(pos) or ori_cs .. suffix
    require('Comment.ft').set(lang, sub)
    require('Comment.api').insert.linewise.eol()
    require('Comment.ft').set(lang, cs)
  end
end
-- current line
vim.keymap.set('n', 'gva', comment_eol('[!code ++]'))
vim.keymap.set('n', 'gvd', comment_eol('[!code --]'))
vim.keymap.set('n', 'gvh', comment_eol('[!code highlight]'))
vim.keymap.set('n', 'gvf', comment_eol('[!code focus]'))
vim.keymap.set('n', 'gvw', comment_eol('[!code warning]'))
vim.keymap.set('n', 'gve', comment_eol('[!code error]'))

---@param action fun()
local function foreach_line(action)
  return function()
    local start_line = vim.fn.line('v')
    local end_line = vim.fn.line('.')
    vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes('<ESC>', true, false, true), 'x', false)
    for i = math.min(start_line, end_line), math.max(start_line, end_line) do
      vim.cmd(tostring(i))
      action()
      vim.api.nvim_feedkeys(esc, 'x', false)
    end
  end
end

-- selection
vim.keymap.set('x', 'vva', foreach_line(comment_eol('[!code ++]')), { desc = '[!code ++]' })
vim.keymap.set('x', 'vvd', foreach_line(comment_eol('[!code --]')), { desc = '[!code --]' })
vim.keymap.set('x', 'vvh', foreach_line(comment_eol('[!code highlight]')), { desc = '[!code highlight]' })
vim.keymap.set('x', 'vvf', foreach_line(comment_eol('[!code focus]')), { desc = '[!code focus]' })
vim.keymap.set('x', 'vvw', foreach_line(comment_eol('[!code warning]')), { desc = '[!code warning]' })
vim.keymap.set('x', 'vve', foreach_line(comment_eol('[!code error]')), { desc = '[!code error]' })

```
